### PR TITLE
Enhancing Perf Framework with Quic protocol support

### PR DESCRIPTION
**Pull request by pavan187**
**What type of PR is this?**
 /kind feature

**What this PR does / why we need it**:
This PR enhance the performance framework with quic protocol test support.
Now user can test against each protocol just by giving config item while performing the test.

**Which issue(s) this PR fixes**:
Fixes #NA 

**Special notes for your reviewer**:
/cc @edisonxiang @sids-b @rohitsardesai83 @m1093782566

### CODE REVIEW

File: tests/e2e/utils/common.go [hunk 1]
```diff
@@ -124,6 +126,12 @@ func edgecoreDeploymentSpec(imgURL, configmap string, replicas int) *apps.Deploy
 
 //Function to create cloudcore deploymentspec object
 func cloudcoreDeploymentSpec(imgURL, configmap string, replicas int) *apps.DeploymentSpec {
+	var portInfo []v1.ContainerPort
+	if !ProtocolQuic{
+		portInfo = []v1.ContainerPort{{ContainerPort: 10000, Protocol: "TCP", Name: "cloudhub"}}
```
Comment (m1093782566) [inline tests/e2e/utils/common.go]: 10000 is for websocket protocol right?

File: tests/e2e/utils/common.go [hunk 2]
```diff
@@ -37,7 +37,9 @@ import (
 const (
 	Namespace = "default"
 )
-
+var (
+	ProtocolQuic	bool
```
Comment (m1093782566) [inline tests/e2e/utils/common.go]: Protocol WebSocket please.

File: tests/e2e/utils/common.go (hunk 1 above)
Comment (pavan187) [reply tests/e2e/utils/common.go]: @m1093782566 , Yes

File: tests/e2e/utils/common.go (hunk 2 above)
Comment (pavan187) [reply tests/e2e/utils/common.go]: OK

File: tests/performance/assets/01-configmap.yaml [hunk 3]
```diff
@@ -10,21 +10,34 @@ data:
...
     controller:
       kube:
-        master: http://192.168.20.42:8080
+        master: http://192.168.20.128:8080
         namespace: ""
         content_type: "application/vnd.kubernetes.protobuf"
         qps: 5
         burst: 10
         node_update_frequency: 10
+        kubeconfig: ""   #Enter path to kubeconfig file to enable https connection to k8s apiserver
```
Comment (sids-b) [inline tests/performance/assets/01-configmap.yaml]: I think we can delete this line as we are specifying empty path

File: tests/performance/assets/01-configmap.yaml [hunk 4]
```diff
@@ -10,21 +10,34 @@ data:
   controller.yaml: |
     controller:
       kube:
-        master: http://192.168.20.42:8080
+        master: http://192.168.20.128:8080
```
Comment (sids-b) [inline tests/performance/assets/01-configmap.yaml]: We need tocomeup with a way soon to mask IP's

File: tests/performance/assets/01-configmap.yaml [hunk 5]
```diff
@@ -10,21 +10,34 @@ data:
...
-      node-limit: 100
+      node-limit: 10
+    devicecontroller:
+      kube:
+        master: http://192.168.20.128:8080
+        namespace: ""
+        content_type: "application/vnd.kubernetes.protobuf"
+        qps: 5
+        burst: 10
+        kubeconfig: ""
```
Comment (sids-b) [inline tests/performance/assets/01-configmap.yaml]: I think we can delete this line as we are specifying empty path

File: tests/performance/assets/02-edgeconfigmap.yaml [hunk 6]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: edgecore-configmap-rov
+  name: edgecore-configmap-34s
```
Comment (sids-b) [inline tests/performance/assets/02-edgeconfigmap.yaml]: what is 34s ? Just asking out of curiosity

File: tests/e2e/utils/config.go [hunk 7]
```diff
@@ -43,6 +43,7 @@ type Config struct {
 	EdgeImageUrl                   string            `json:"edgeimageurl"`
 	Namespace                      string            `json:"namespace"`
 	ControllerStubPort             int               `json:"controllerstubport"`
+	Protocol 		       string            `json:"protocol"`
```
Comment (rohitsardesai83) [inline tests/e2e/utils/config.go]: please run gofmt

File: tests/performance/assets/01-configmap.yaml [hunk 8]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: cloudcore-configmap-j8rzh
+  name: cloudcore-configmap-sli0c
```
Comment (rohitsardesai83) [inline tests/performance/assets/01-configmap.yaml]: can we avoid such complex names ? on what basis are we doing this name change ?

File: tests/performance/assets/02-edgeconfigmap.yaml [hunk 9]
```diff
@@ -17,24 +17,34 @@ data:
...
             write-deadline: 15 # second
             read-deadline: 15 # second
         controller:
+            protocol: quic
             placement: false
             heartbeat: 15  # second
             refresh-ak-sk-interval: 10 # minute
             auth-info-files-path: /var/IEF/secret
-            placement-url: wss://192.168.20.107:31765/e632aba927ea4ac2b575ec1603d56f10/perf-node-ymysm0j2yw/events
+            placement-url: https://10.154.193.32:7444/v1/placement_external/message_queue
```
Comment (rohitsardesai83) [inline tests/performance/assets/02-edgeconfigmap.yaml]: are we using placement url ?

**Comment by edisonxiang** [comment]
Thanks @pavan187 I will review it once another commit is pushed.

File: tests/e2e/utils/common.go [hunk 10]
```diff
@@ -368,12 +376,18 @@ func ExposeCloudService(name, serviceHandler string) error {
 
 //CreateServiceObject function to create a servcice object
 func CreateServiceObject(name string) *v1.Service {
+	var portInfo []v1.ServicePort
+	if !ProtocolQuic{
+		portInfo = []v1.ServicePort{{Protocol: "TCP", Port: 10000, TargetPort: intstr.FromInt(10000)}}
+	}else {
+		portInfo = []v1.ServicePort{{Protocol: "UDP", Port: 10001, TargetPort: intstr.FromInt(10001)}}
```
Comment (ugvddm) [inline tests/e2e/utils/common.go]: devil number, it's sound not good

Comment (pavan187) [reply tests/e2e/utils/common.go]: @ugvddm , Thank you for reviewing the PR, can you please elobarate you comment, i could not understand your meaning of "devil number"...

Comment (ugvddm) [reply tests/e2e/utils/common.go]: It's better to define a constant instance of using number in code

Comment (m1093782566) [reply tests/e2e/utils/common.go]: > It's better to define a constant instance of using number in code

+1

**Comment by sids-b** [comment]
@pavan187 , just a few nit comments are pending. Overall looks good to me.

cc @edisonxiang for review

File: tests/performance/scripts/runperf.sh [hunk 11]
```diff
@@ -35,10 +35,11 @@ cat >config.json<<END
         "node_num": 10,
         "imagerepo": "kubeedge",
         "k8smasterforprovisionedgenodes": "$K8SMasterForProvisionEdgeNodes",
-        "cloudimageurl": "kubeedge/edgecontroller-test:v2.1",
-        "edgeimageurl": "kubeedge/edgecore-test:v2.1",
+        "cloudimageurl": "pavan187/cloudcore:v2.2",
```
Comment (edisonxiang) [inline tests/performance/scripts/runperf.sh]: we'd better to upload to kubeedge, but we can update later when we have v0.3 version:)

**Comment by edisonxiang** [comment]
Let's merge it:)
Thanks @pavan187 
/approve

**Pull request merged**